packages/plugin-compat/sources/extensions.ts

```diff
@@ -52,4 +52,12 @@ export const packageExtensions: Array<[string, any]> = [
       [`@types/keyv`]: `^3.1.1`,
     },
   }],
+  // vscode-jsonrpc and vscode-languageserver-protocol are imported at runtime
+  // yet absent from typescript-language-server's dependencies
+  [`typescript-language-server@*`, {
+    dependencies: {
+      [`vscode-jsonrpc`]: `^5.0.1`,
```

Comment:
Are you sure for `vscode-jsonrpc`? From a quick look to their repo, it seems to only be used for type imports, so it shouldn't be imported at runtime 🤔

Reply:
It's required from within `lib/lsp-protocol.calls.proposed.js`, the call stack is as follows.
```
- <project>/.yarn/cache/typescript-language-server-npm-0.4.0-c491a3c644-2.zip/node_modules/typescript-language-server/lib/lsp-protocol.calls.proposed.js
- <project>/.yarn/cache/typescript-language-server-npm-0.4.0-c491a3c644-2.zip/node_modules/typescript-language-server/lib/lsp-connection.js
- <project>/.yarn/cache/typescript-language-server-npm-0.4.0-c491a3c644-2.zip/node_modules/typescript-language-server/lib/cli.js
```